src/ext/BalExtension/wixstdba/WixStandardBootstrapperApplication.cpp

```diff
@@ -807,15 +807,15 @@ class CWixStandardBootstrapperApplication : public CBalBaseBootstrapperApplicati
                 }
 
                 wz = sczFormattedString ? sczFormattedString : pPackage->sczDisplayName ? pPackage->sczDisplayName : wzPackageId;
-            }
 
-            // Needs to match MsiEngineCalculateInstallUiLevel in msiengine.cpp in Burn.
-            BOOTSTRAPPER_ACTION_STATE packageAction = fExecute ? pPackage->executeAction : pPackage->rollbackAction;
-            fShowingInternalUiThisPackage = pPackage && pPackage->fDisplayInternalUI &&
-                                            BOOTSTRAPPER_ACTION_UNINSTALL != packageAction &&
-                                            BOOTSTRAPPER_ACTION_REPAIR != packageAction &&
-                                            (BOOTSTRAPPER_DISPLAY_FULL == m_command.display ||
-                                            BOOTSTRAPPER_DISPLAY_PASSIVE == m_command.display);
+                // Needs to match MsiEngineCalculateInstallUiLevel in msiengine.cpp in Burn.
+                BOOTSTRAPPER_ACTION_STATE packageAction = fExecute ? pPackage->executeAction : pPackage->rollbackAction;
+                fShowingInternalUiThisPackage = pPackage && pPackage->fDisplayInternalUI &&
```

Comment:
This looks right, not sure how I made this mistake. You could do an optimization here since now we know `pPackage` is not `NULL`.